catowseriOS/catowser/ServiceRegistry/UIServiceRegistry.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  UIServiceRegistry.swift
+//  catowser
+//
+//  Created by Andrey Ermoshin on 19.09.2024.
+//  Copyright © 2024 Cotton (Catowser). All rights reserved.
+//
+
+import CoreBrowser
+import UIKit
+
+/// A service registry used only as a main actor for UI
+/// to be able to use it in SwiftUI or UIKit views for observing
+/// because even if it is a global state it makes it complicated
+/// if it is a global actor.
+@MainActor final class UIServiceRegistry {
+    /// Shared instance
+    static func shared() -> UIServiceRegistry {
+        if let holder = internalInstance {
+            return holder
+        }
+
+        let created = UIServiceRegistry(DefaultTabProvider.shared)
+        internalInstance = created
+        return created
+    }
+    
+    /// Internal instance for the singletone
+    static private var internalInstance: UIServiceRegistry?
+    
+    /// Default positioning settings
+    private let positioning: TabsStates
+    /// A workaround to be able to use available marker.
+    ///
+    /// Should be main actor data beacuse observed almost
+    /// everytime in some view controller.
+    private var _tabsSubject: Any? = nil
```

Comment:
double check if it is possible to mark it as main actor as well? The property holder (UIServiceRegistry) is already a main actor tho